packages/core-users/src/module/configureUsersModule.ts

```diff
@@ -198,13 +198,32 @@ export const configureUsersModule = async ({
 
     updateProfile: async (_id, profile, userId) => {
```

Comment:
The customFields was a half-baken solution which was not clear what to expect from.

I agree with you that if you use "extension" of input parameters, the actual update profile should only update fields that have been set but leave the others as is.

But like I stated above, that would not be consistent: We should propably transition to a way where we set an arbitrary "meta" field which could be anything JSON-stringifiable. Then we only have 1 variable to set and because it's arbitrary it's clear for the API user that if he sets meta, he overwrites what was meta before.

So I'd keep the simpler method as before, if meta is sent, overwrite meta.